derive-encode/src/lib.rs

```diff
@@ -132,6 +184,77 @@ pub fn derive_encode_label_value(input: TokenStream) -> TokenStream {
     gen.into()
 }
 
+#[derive(Clone)]
+enum ValueCase {
+    Lower,
+    Upper,
+    NoChange,
+}
+
+impl ValueCase {
+    fn apply(&self, ident: &Ident) -> Ident {
+        match self {
+            ValueCase::Lower => Ident::new(&ident.to_string().to_lowercase(), ident.span()),
+            ValueCase::Upper => Ident::new(&ident.to_string().to_uppercase(), ident.span()),
+            ValueCase::NoChange => ident.clone(),
+        }
+    }
+}
+
+struct LabelConfig {
+    value_case: ValueCase,
+}
+
+impl Default for LabelConfig {
+    fn default() -> Self {
+        Self {
+            value_case: ValueCase::NoChange,
+        }
+    }
+}
+
+impl Parse for LabelConfig {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let mut config = LabelConfig::default();
+
+        while input.peek(Ident) {
+            let ident: Ident = input.parse()?;
+
+            match ident.to_string().as_str() {
+                "value_case" => {
+                    let _: Token![=] = input.parse()?;
+                    let case: LitStr = input.parse()?;
+
+                    match case.value().as_str() {
+                        "lower" => config.value_case = ValueCase::Lower,
+                        "upper" => config.value_case = ValueCase::Upper,
+                        invalid => {
+                            return Err(syn::Error::new(
+                                case.span(),
+                                format!(
+                                "value case may only be \"lower\" or \"upper\", not \"{invalid}\""
+                            ),
+                            ))
```

Comment:
To de-duplicate this code with the one above in line 146, how about implementing `FromStr` for `ValueCase`?

Reply:
It will take me some time, but I think I can do this